src/AutoMapper/PropertyMap.cs

```diff
@@ -219,7 +219,24 @@ internal void Seal(TypeMapRegistry typeMapRegistry)
             }
 
             //_mapperFunc = (mappedObject, context) => DestinationProperty.SetValue(mappedObject, valueResolverFunc(mappedObject, context));
-            Expression mapperExpr = Expression.Assign(destMember, Expression.Convert(valueResolverExpr, DestinationPropertyType));
+            Expression mapperExpr;
+            if (DestinationProperty.MemberInfo is FieldInfo)
+            {
+                mapperExpr = Expression.Assign(destMember, Expression.Convert(valueResolverExpr, DestinationPropertyType));
+            }
+            else
+            {
+                var prop = (PropertyInfo) DestinationProperty.MemberInfo;
+                var setter = prop.GetSetMethod(true);
+                if (setter == null)

```

Comment:
Stupid question, but don't you filter out properties by if they have a setter somewhere else in code?